src/composed/signed_key/key_parser.rs

```diff
@@ -0,0 +1,229 @@
+use crate::errors::Result;
+use crate::packet::{self, Packet, Signature, SignatureType, UserAttribute, UserId};
+use crate::types::{KeyVersion, PublicKeyTrait, SignedUser, SignedUserAttribute, Tag};
+use crate::{SignedKeyDetails, SignedPublicSubKey, SignedSecretSubKey};
+
+#[allow(clippy::complexity)]
+pub fn next<I, IKT>(
+    packets: &mut std::iter::Peekable<I>,
+    key_tag: Tag,
+    parse_secrect_subkeys: bool,
+) -> Option<
+    Result<(
+        IKT,
```

Comment:
I'm unsure how to think about this generic.

I think the `IKT` should definitely never be a subkey type. Shouldn't it always be a `packet::PublicKey`?

Reply:
Ah. Or a `packet::SecretKey`?

Still, it seems to me the type system isn't currently guaranteeing that we won't get a subkey packet in this location? ... which would be a bit sad.

Reply:
I am confused, it is quite strictly reduced type wise in the where clause, what are you worried about?

Reply:
I was wondering if it'd be possible and useful to disallow subkey types in this place. But I have no strong feelings about this.